ocs_ci/utility/networking.py

```diff
@@ -0,0 +1,85 @@
+"""
+Module that contains network related functions
+"""
+
+import ipaddress
+import logging
+
+from ocs_ci.framework import config
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.exceptions import (
+    DeploymentPlatformNotSupported,
+    UnavailableResourceException,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+def annotate_worker_nodes_with_mon_ip():
+    """
+    Annotate worker nodes with annotation 'network.rook.io/mon-ip: <IPAddress>'
+    """
+    if not config.ENV_DATA["platform"].lower() in constants.BAREMETAL_PLATFORMS:
+        raise DeploymentPlatformNotSupported(
```

Comment:
The `DeploymentPlatformNotSupported` is quite specific and actually used only on one other place in the framework, where it is raised for unsupported platform for deployment, in this case I would propose `UnexpectedDeploymentConfiguration` or `ConfigurationError` exceptions.